docs/configuration.md

```diff
@@ -0,0 +1,128 @@
+    .withDelegates(Arrays.asList("delegate@project.iam.googleapis.com"))
+    .withGoogleCredentials(myCredentials)
+    .build();
+
+    ConnectorRegistry.register("my-connector",config);
+```
+
+Then the application would use the connector by adding
+the `cloudSqlNamedConnector`  to the JDBC URL:
+
+```java
+String jdbcUrl="jdbc:mysql:///<DATABASE_NAME>?"+
+    +"cloudSqlInstance=project:region:instance"
+    +"&cloudSqlNamedConnector=my-connector"
+    +"&socketFactory=com.google.cloud.sql.mysql.SocketFactory"
+    +"&user=<DB_USER>&password=<PASSWORD>";
+```
+
+Or using JDBC connection properties, by adding the `cloudSqlNamedConnector`
+property:
+
+```java
+// Set up URL parameters
+Properties connProps=new Properties();
+
+    connProps.setProperty("user","<DB_USER>");
+    connProps.setProperty("password","<PASSWORD>");
+    connProps.setProperty("sslmode","disable");
+    connProps.setProperty("socketFactory","<DRIVER_CLASS>");
+    connProps.setProperty("cloudSqlInstance","project:region:instance");
+
+    connProps.setProperty("cloudSqlNamedConnector","my-connector");
+
+// Initialize connection pool
+    HikariConfig config=new HikariConfig();
+    config.setJdbcUrl("jdbc:mysql:///<DB_NAME>");
+    config.setDataSourceProperties(connProps);
+    config.setConnectionTimeout(10000); // 10s
+
+    HikariDataSource connectionPool=new HikariDataSource(config);
+```
+
+When using a named connector, the JDBC connection will ignore all connector
+configuration properties in the JDBC connection properties, using the named
+connector configuration instead. See the full list
+of [connector configuration properties](#connector-configuration) below.
+
+### Closing Named Connectors
+
+You may close a named connector. This will cause all subsequent attempts to
+connect using this named connector to fail. This will stop the background
+processes to refresh the connector's credentials.
+
+```java
+ConnectorRegistry.close("my-connector");
+```
+
+### Shutdown The Connector Registry
+
+The application may shutdown the ConnectorRegistry, preventing any new
+connections and stopping all connector threads.
+
+```java
+ConnectorRegistry.shutdown();
+```
+
+After calling `ConnectorRegistry.shutdown()`, the next attempt to connect to a
+database using a SocketFactory or R2DBC ConnectionFactory, or
+to `ConnectorRegistry.register()` will start a new connector registry.
+
+### Configuring Google Credentials
+
+TODO
+
+- `Supplier<GoogleCredentials> googleCredentialsSupplier` - a supplier to
+  provide
+  `GoogleCredentials` instances to the connector.
+- `GoogleCredentials googleCredentials` - an instance of GoogleCredentials to
+  use
+  for all requests.
+- `String googleCredentialsPath` - path to file containing GoogleCredentials
+  json. May be set with the JDBC Connection
+  Property `cloudSqlGoogleCredentialsPath`.
+
+Users should only set one field. If more than one field is set, the connector 
```

Comment:
I think we should throw an exception and just not support setting multiple credentials per connector.

Reply:
Fixed. The code and docs are updated to throw an exception.